dokka-runners/dokka-gradle-plugin/src/testFunctional/kotlin/DokkaGeneratorFailureTest.kt

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2014-2024 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package org.jetbrains.dokka.gradle
+
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.string.shouldContain
+import org.jetbrains.dokka.gradle.internal.DokkaConstants.DOKKA_VERSION
+import org.jetbrains.dokka.gradle.utils.*
+
+class DokkaGeneratorFailureTest : FunSpec({
+    context("DokkaGenerator failure:") {
+        val project = gradleKtsProjectTest("dokka-generator-failure") {
```

Comment:
Can you make this test more straightforward? Or even add one more with a custom exception/pre-generation check?
For example, if intersecting roots are allowed in the future, this case could be lost.
See [the original test ](https://github.com/Kotlin/dokka/blob/e59dc2a61b1c94e1a53c27555899392377e35f3c/dokka-runners/dokka-gradle-plugin/src/classicTest/kotlin/AutomagicProxyTest.kt#L32C66-L32C94)